main/cropper.js

```diff
@@ -16,66 +16,134 @@ const prodPath = formatUrl({
+    croppers.delete(id);
+  }
 
-    if (isDev) {
-      cropper.openDevTools({mode: 'detach'});
-    }
+  screen.removeAllListeners('display-removed');
+  screen.removeAllListeners('display-added');
+};
 
-    cropper.on('blur', () => {
-      if (!shouldIgnoreBlur && !cropper.webContents.isDevToolsFocused()) {
-        cropper.close();
-      }
-    });
+const openCropper = (display, activeDisplayId) => {
+  const {id, bounds} = display;
+  const {x, y, width, height} = bounds;
 
-    cropper.on('closed', () => {
-      cropper = null;
-    });
+  const cropper = new BrowserWindow({
+    x,
+    y,
+    width,
+    height,
+    hasShadow: false,
+    enableLargerThanScreen: true,
+    resizable: false,
+    moveable: false,
+    frame: false,
+    transparent: true,
+    show: false
+  });
+
+  cropper.loadURL(url);
+  cropper.setAlwaysOnTop(true, 'screen-saver', 1);
+
+  cropper.webContents.on('did-finish-load', () => {
+    const isActive = activeDisplayId === id;
+    cropper.webContents.send('display', {id, x, y, width, height, isActive});
+  });
+
+  if (isDev) {
+    cropper.openDevTools({mode: 'detach'});
   }
+
+  cropper.on('closed', closeAllCroppers);
+  croppers.set(id, cropper);
+  return cropper;
 };
 
-const ignoreBlur = () => {
-  shouldIgnoreBlur = true;
+const openCropperWindow = () => {
+  closeAllCroppers();
+
+  const {screen} = electron;
+  const displays = screen.getAllDisplays();
+  const activeDisplayId = screen.getDisplayNearestPoint(screen.getCursorScreenPoint()).id;
+
+  for (const display of displays) {
+    openCropper(display, activeDisplayId);
+  }
+
+  for (const cropper of croppers.values()) {
+    cropper.showInactive();
+  }
+
+  croppers.get(activeDisplayId).focus();
+
+  screen.on('display-removed', (event, oldDisplay) => {
+    const {id} = oldDisplay;
+    const cropper = croppers.get(id);
+
+    const wasFocused = cropper.isFocused();
+
+    cropper.removeAllListeners('closed');
+    cropper.destroy();
+    delete croppers[id];
+
+    if (wasFocused) {
+      const activeDisplayId = screen.getDisplayNearestPoint(screen.getCursorScreenPoint()).id;
+      croppers.get(activeDisplayId).focus();
+    }
+  });
+
+  screen.on('display-added', (event, newDisplay) => {
+    const cropper = openCropper(newDisplay);
+    cropper.showInactive();
+  });
 };
 
-const restoreBlur = () => {
-  shouldIgnoreBlur = false;
-  cropper.focus();
+const preventDefault = event => event.preventDefault();
+
+const selectApp = async (window, activateWindow) => {
+  for (const cropper of croppers.values()) {
+    cropper.prependListener('blur', preventDefault);
+  }
+
+  await activateWindow(window.ownerName);
+
+  const {screen} = electron;
+  const {x, y, width, height, ownerName} = window;
+
+  const display = screen.getDisplayMatching({x, y, width, height});
+  const {id, bounds: {x: screenX, y: screenY}} = display;
+
+  // For some reason this happened a bit too early without the timeout
+  setTimeout(() => {
```

Comment:
Would be nicer to use [`delay`](https://github.com/sindresorhus/delay) here. Then we don't have to nest.